redb-derive/tests/derive_tests.rs

```diff
@@ -0,0 +1,202 @@
+        format!("{:?}", TupleStruct::type_name())
+    );
+}
+
+#[test]
+fn test_single_field_type_name() {
+    let type_name = SingleField::type_name();
+    assert_eq!(
+        format!("{:?}", type_name),
+        format!("{:?}", SingleField::type_name())
+    );
+}
+
+#[test]
+fn test_mixed_types_type_name() {
+    let type_name = MixedTypes::type_name();
+    assert_eq!(
+        format!("{:?}", type_name),
+        format!("{:?}", MixedTypes::type_name())
+    );
+}
+
+#[test]
+fn test_simple_struct_serialization() {
+    let original = SimpleStruct {
+        id: 42,
+        name: "test".to_string(),
+    };
+
+    let bytes = SimpleStruct::as_bytes(&original);
+    let deserialized = SimpleStruct::from_bytes(bytes.as_ref());
+
+    assert_eq!(deserialized.id, 42);
+    assert_eq!(deserialized.name, "test");
+}
+
+#[test]
+fn test_tuple_struct_serialization() {
+    let original = TupleStruct(123, true);
+
+    let bytes = TupleStruct::as_bytes(&original);
+    let deserialized = TupleStruct::from_bytes(bytes.as_ref());
+
+    assert_eq!(deserialized.0, 123);
+    assert_eq!(deserialized.1, true);
+}
+
+#[test]
+fn test_single_field_serialization() {
+    let original = SingleField { value: -456 };
+
+    let bytes = SingleField::as_bytes(&original);
+    let deserialized = SingleField::from_bytes(bytes.as_ref());
+
+    assert_eq!(deserialized.value, -456);
+}
+
+#[test]
+fn test_mixed_types_serialization() {
+    let original = MixedTypes {
+        fixed: 1000,
+        variable: "hello world".to_string(),
+        optional: Some(255),
+    };
+
+    let bytes = MixedTypes::as_bytes(&original);
+    let deserialized = MixedTypes::from_bytes(bytes.as_ref());
+
+    assert_eq!(deserialized.fixed, 1000);
+    assert_eq!(deserialized.variable, "hello world");
+    assert_eq!(deserialized.optional, Some(255));
+}
+
+#[test]
+fn test_fixed_width() {
+    assert_eq!(SingleField::fixed_width(), Some(4));
+    assert_eq!(SimpleStruct::fixed_width(), None);
+    assert_eq!(TupleStruct::fixed_width(), Some(9));
+    assert_eq!(MixedTypes::fixed_width(), None);
+}
+
+#[test]
+fn test_database_integration() {
+    let tmpfile = create_tempfile();
+    let db = Database::create(tmpfile.path()).unwrap();
+
+    let table_def: TableDefinition<u32, SimpleStruct> = TableDefinition::new("test_table");
+
+    let write_txn = db.begin_write().unwrap();
+    {
+        let mut table = write_txn.open_table(table_def).unwrap();
+        let value = SimpleStruct {
+            id: 100,
+            name: "database test".to_string(),
+        };
+        table.insert(&1, &value).unwrap();
+    }
+    write_txn.commit().unwrap();
+
+    let read_txn = db.begin_read().unwrap();
+    let table = read_txn.open_table(table_def).unwrap();
+    let retrieved = table.get(&1).unwrap().unwrap();
+
+    assert_eq!(retrieved.value().id, 100);
+    assert_eq!(retrieved.value().name, "database test");
+}
+
+#[test]
+fn test_tuple_struct_database_integration() {
```

Comment:
Add a test with a more complex case. Like having the struct inside a vec. e.g. `TableDefinition<u32, Vec<TupleStruct>>`